packages/rspack/src/compiler.ts

```diff
@@ -374,6 +374,7 @@ class Compiler {
 	}
 
 	#newCompilation(native: binding.JsCompilation) {
+		this.hooks.compile.call();
```

Comment:
The `call` timing is not right, you could refer https://github.com/webpack/webpack/blob/d96ce76f4d3996f32367b4b4231039451550c1f7/lib/Compiler.js#L1161-L1203 and https://github.com/webpack/webpack/blob/d96ce76f4d3996f32367b4b4231039451550c1f7/lib/Compiler.js#L1117-L1124, 
the correct timing is 
```mermaid
flowchart LR
   compile-. newCompilation .->thisCompilation-->compilation
```